File: app/src/main/java/ashcollauthlogin/darkzek/com/IntroActivity.java
package ashcollauthlogin.darkzek.com;

import android.content.Intent;
import android.content.SharedPreferences;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.KeyEvent;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

public class IntroActivity extends AppCompatActivity {

    private ViewPager mViewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_intro);

        mViewPager = findViewById(R.id.viewpager);

        // Set an Adapter on the ViewPager
        mViewPager.setAdapter(new IntroAdapter(getSupportFragmentManager()));

        // Set a PageTransformer
        mViewPager.setPageTransformer(false, new IntroPageTransformer());

        TabLayout tabLayout = findViewById(R.id.tabDots);
        tabLayout.setupWithViewPager(findViewById(R.id.viewpager), true);

        findViewById(R.id.skip).setOnClickListener(view -> mViewPager.setCurrentItem(3));
        findViewById(R.id.next).setOnClickListener(view -> NextButtonClicked(view));


    }

    public void NextButtonClicked(View v) {
        int page = mViewPager.getCurrentItem();

        if (page != 2) {
            //Increase page
            mViewPager.setCurrentItem(page += 1);
            Log.i("SETTING", page + "");
            return;
        }

        onContinueClick(v);
    }

    public void onContinueClick(View v)
    {
        String usernameText = ((EditText)findViewById(R.id.usernameField)).getText().toString();
        String passwordText = ((EditText)findViewById(R.id.passwordField)).getText().toString();

        //Check if a username was put in
        if (usernameText.equalsIgnoreCase("")) {
            //Invalid!
            ((EditText)findViewById(R.id.usernameField)).setError("This field can not be blank!");
            return;
        }

        if (passwordText.equalsIgnoreCase("")) {
            //Invalid!
            ((EditText)findViewById(R.id.passwordField)).setError("This field can not be blank!");
            return;
        }

        SaveCredentials(usernameText, passwordText);

        Log.i("Started", "Started");

        String user = getSharedPreferences("credentials", MODE_PRIVATE).getString("username", "DEFAULT");

        //Show toast
        Toast.makeText(this, "Logging in as " + user,
                Toast.LENGTH_LONG).show();

        LoadSettingsActivity();
    }

    public void SaveCredentials(String username, String password) {
        SharedPreferences.Editor editor = getSharedPreferences("credentials", MODE_PRIVATE).edit();
        editor.putString("username", username);
        editor.putString("password", password);
        editor.putInt("timesLoggedIn", 0);
        editor.commit();
    }


    public void LoadSettingsActivity() {
        Intent myIntent = new Intent(this, SettingsActivity.class);
        startActivity(myIntent);

        finish();
    }
}

File: app/src/main/java/ashcollauthlogin/darkzek/com/IntroActivity.kt
package ashcollauthlogin.darkzek.com

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.support.design.widget.TabLayout
import android.support.v4.view.ViewPager
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.KeyEvent
import android.view.View
import android.widget.EditText
import android.widget.Toast

class IntroActivity : AppCompatActivity() {

    private var mViewPager: ViewPager? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_intro)

        mViewPager = findViewById(R.id.viewpager)

        // Set an Adapter on the ViewPager
        mViewPager!!.adapter = IntroAdapter(supportFragmentManager)

        // Set a PageTransformer
        mViewPager!!.setPageTransformer(false, IntroPageTransformer())

        val tabLayout = findViewById<TabLayout>(R.id.tabDots)
        tabLayout.setupWithViewPager(findViewById(R.id.viewpager), true)

        findViewById<View>(R.id.skip).setOnClickListener { view -> mViewPager!!.currentItem = 3 }
        findViewById<View>(R.id.next).setOnClickListener { view -> NextButtonClicked(view) }


    }

    fun NextButtonClicked(v: View) {
        var page = mViewPager!!.currentItem

        if (page != 2) {
            //Increase page
            mViewPager!!.currentItem = page + 1
            return
        }

        onContinueClick(v)
    }

    fun onContinueClick(v: View) {
        val usernameText = (findViewById<View>(R.id.usernameField) as EditText).text.toString()
        val passwordText = (findViewById<View>(R.id.passwordField) as EditText).text.toString()

        //Check if a username was put in
        if (usernameText.equals("", ignoreCase = true)) {
            //Invalid!
            (findViewById<View>(R.id.usernameField) as EditText).error = "This field can not be blank!"
            return
        }

        if (passwordText.equals("", ignoreCase = true)) {
            //Invalid!
            (findViewById<View>(R.id.passwordField) as EditText).error = "This field can not be blank!"
            return
        }

        SaveCredentials(usernameText, passwordText)

        Log.i("Started", "Started")

        val user = getSharedPreferences("credentials", Context.MODE_PRIVATE).getString("username", "DEFAULT")

        //Show toast
        Toast.makeText(this, "Logging in as " + user!!,
                Toast.LENGTH_LONG).show()

        LoadSettingsActivity()
    }

    fun SaveCredentials(username: String, password: String) {
        val editor = getSharedPreferences("credentials", Context.MODE_PRIVATE).edit()
        editor.putString("username", username)
        editor.putString("password", password)
        editor.putInt("timesLoggedIn", 0)
        editor.commit()
    }


    fun LoadSettingsActivity() {
        val myIntent = Intent(this, MainActivity::class.java)
        startActivity(myIntent)

        finish()
    }
}
